s3stream/src/main/java/com/automq/stream/s3/operator/AbstractObjectStorage.java

```diff
@@ -148,20 +149,55 @@ public CompletableFuture<ByteBuf> rangeRead(ReadOptions options, String objectPa
             return cf;
         }
 
-        TimerUtil timerUtil = new TimerUtil();
-        networkInboundBandwidthLimiter.consume(options.throttleStrategy(), end - start).whenComplete((v, ex) -> {
-            NetworkStats.getInstance().networkLimiterQueueTimeStats(AsyncNetworkBandwidthLimiter.Type.INBOUND, options.throttleStrategy())
-                .record(timerUtil.elapsedAs(TimeUnit.NANOSECONDS));
+        LongFunction<CompletableFuture<Void>> networkInboundBandwidthLimiterFunction =
```

Comment:
When end == RANGE_READ_TO_READ: we can use size = 1 in rangeRead to obtain a rate-limiting token for rate limiting. Then after the reading is completed, use the ThrottleStrategy.BYPASS strategy to consume size = the actual returned size - 1.

Reply:
got it I will change the code later